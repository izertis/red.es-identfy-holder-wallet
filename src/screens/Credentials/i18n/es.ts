const localeES = {
	TITLE: 'Credenciales',
	PlACEHOLDER_TITLE: 'Te damos la bienvenida',
	PlACEHOLDER_TEXT:
		'En esta cartera podrás guardar tus credenciales de forma segura. Para obtener una credencial, simplemente solicítala a la organización correspondiente y recibirás por correo electrónico la información necesaria. Luego, puedes guardarla directamente en la aplicación o escanear un código QR para añadirla rápidamente.',
	BUTTON_SCAN: 'Escanear',
	BUTTON_MAILBOX: 'Actualizar credenciales',
	BUTTON_MAILBOX_SUCCESS: 'El listado de credenciales se ha actualizado',
	CREDENTIAL_DETAILS_ISSUER: 'Organización: ',
	CREDENTIAL_DETAIL_TYPE: 'Credencial: ',
	CREDENTIAL_DETAIL_STATUS: 'Estado: ',
	ISSUANCE_DATE: 'Fecha emisión: ',
	EXPIRES: 'Expira: ',
	NEVER: 'Nunca',
}

export default localeES
