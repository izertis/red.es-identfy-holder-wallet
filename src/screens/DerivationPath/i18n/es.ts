const localeES = {
  TITLE: 'Forma de derivación',
  SUBTITLE:
    'Este número te permite recuperar tu cuenta en caso de que olvides el acceso. Recomendamos que la memorices o copies en un lugar seguro.  Por seguridad, nunca la reveles a otra persona.',
  HIDDEN_TITLE: 'Pulse aquí para revelar su forma de derivación',
  LONG_PRESS: 'Manten presionada para copiarla',
  BUTTON: 'Continuar',
  GO_BACK: 'Volver',
}

export default localeES
