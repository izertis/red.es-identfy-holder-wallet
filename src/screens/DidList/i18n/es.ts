const localeES = {
	TITLE: 'Mis identidades',
	CARD_DESCRIPTION: 'DID registrado en {{network}} de {{entity}}',
	PRESS_TO_COPY: 'Manten presionado el DID para copiarlo',
	ADD_NEW_DID: 'Agregar DID',
	NO_DID_CREATED: 'No has creado ningún DID, por favor presiona el botón debajo para añadir DID',
}

export default localeES
