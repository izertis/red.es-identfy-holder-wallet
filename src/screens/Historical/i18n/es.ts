const localeES = {
	TITLE: 'Histórico',
	PLACEHOLDER_TITLE: 'Histórico',
	PLACEHOLDER_TEXT: 'Aquí aparecera un histórico de las acciones que has ejecutado en tu wallet.',
	CREDENTIAL_DETAILS_ISSUER: 'Organización',
	DATE: 'Fecha',
	CREDENTIAL_ADDED: 'Acreditación añadida',
	CREDENTIAL_EXPIRED: 'Acreditación expirada',
	PRESENTATION_EMITTED: 'Presentación realizada',
	CREDENTIAL_REVOKED: 'Credencial revocada',
	PRESENTATION_REVOKED: 'Revocación de acceso',
	DEFAULT_MESSAGE: 'Acción desconocida',
}

export default localeES
