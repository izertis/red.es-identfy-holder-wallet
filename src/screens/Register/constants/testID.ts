export const testID = {
  REGISTER_BUTTON: "register-button",
  FIRST_PIN_INPUT: "first-pin-input",
  SECOND_PIN_INPUT: "second-pin-input",
  TERMS_CHECKBOX: "terms-checkbox",
  REGISTER_ERROR_MESSAGE: "register-error-message",
};
