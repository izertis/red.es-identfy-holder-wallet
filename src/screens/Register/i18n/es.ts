const localeES = {
  TITLE: "Registro",
  SUBTITLE:
    "Añade seguridad a tu aplicación creando un PIN de seguridad",
  PIN_LABEL: "PIN de acceso",
  SECOND_PIN_LABEL: "Repite PIN de acceso",
  LOGIN: "Iniciar sesión",
  PIN: "Crea tu PIN de acceso",
  SECOND_PIN: "Repetir Pin",
  CONDITIONS: "Aceptar términos y condiciones",
  RECOVERY: "Recuperar acceso",
  BUTTON: "Comenzar",
  LOADING: "Creando usuario...",
  ERROR: "Los códigos no son iguales",
  ERROR_PIN_LENGTH: "El PIN ha de tener 4 caracteres",
  ERROR_SAVE: "Error guardando tu DID, inténtalo de nuevo",
  CREATING_DID: "Generando identidad digital...",
  QR_SUBTITLE:
    "Visita la oficina de una organización y solicita que creen tu identidad digital. Te generarán un código QR para que la vincules a tu móvil",
  MESSAGE_ERROR_WALLET_CREATION: "Ha habido un error a la hora de crear tu wallet",
  MESSAGE_SUCCESS_WALLET_CREATION: "Tu wallet ha sido creada exitosamente"
}

export default localeES
