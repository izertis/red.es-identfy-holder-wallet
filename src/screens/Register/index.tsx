import { useContext, useEffect, useState } from 'react'
import { useTranslation } from 'react-i18next'
import { SCREEN } from '../../constants/screens'
import { MessageContext } from '../../context/UserMessage.context'
import LocalStorageService, { STORAGE_KEYS } from '../../services/LocalStorage.service'
import { createEncryptedWallet } from '../../services/Wallet.service'
import { testID } from './constants/testID'
import RegisterI18nKeys from './i18n/keys'
import RegisterStyles from './styles'
import { RootStackScreenProps } from '../../../types'
import TermsModal from '../../components/TermsModal'
import useModal from '../../hooks/useModal'
import { StatusBar } from 'expo-status-bar'
import { getThemeColor, ColorKeys } from '../../constants/Colors'
import { Keyboard } from 'react-native'


const Register = ({ navigation }: RootStackScreenProps<'Register'>) => {
	const { showMessage } = useContext(MessageContext)
	const { t } = useTranslation(SCREEN.Register)

	const [isPinFilled, setIsPinFilled] = useState<boolean>(true)
	const [firstPin, setFirstPin] = useState('')
	const [secondPin, setSecondPin] = useState('')
	const [isLoading, setIsLoading] = useState<boolean>(false)
	const [isTermsCheckboxChecked, setIsTermsCheckboxChecked] = useState(false)
	const { modal, showModal } = useModal()

	useEffect(() => {
		arePinEquals() ? setIsPinFilled(false) : setIsPinFilled(true)
	}, [firstPin, secondPin])

	const createWallet = async () => {
		Keyboard.dismiss()
		try {
			const data = await createEncryptedWallet(firstPin)
			await LocalStorageService.storeBool(STORAGE_KEYS.IS_WALLET_CREATED, true)
			const content = t(RegisterI18nKeys.MESSAGE_SUCCESS_WALLET_CREATION)
			showMessage({ content, type: 'success' })
			setTimeout(() => {
				setIsLoading(false)
				navigation.replace('SecurityPhrase', {
					mneumonic: data.mnemonicPhrase,
					derivationPath: data.derivationPath,
				})
			}, 1000)
		} catch (error) {
			const content = t(RegisterI18nKeys.MESSAGE_ERROR_WALLET_CREATION)
			showMessage({ content, type: 'error' })
			setIsLoading(false)
			console.error('error', error)
		}
	}

	useEffect(() => {
		if (isLoading) {
			if (firstPin.length < 4) {
				const content = t(RegisterI18nKeys.ERROR_PIN_LENGTH)
				showMessage({ content: content, type: 'error' })
				setIsLoading(false)
			} else {
				setTimeout(() => {
					createWallet()
				}, 500)
			}
		}
	}, [isLoading])

	const arePinEquals = () => {
		return firstPin !== '' && secondPin !== '' && firstPin === secondPin
	}

	const renderErrorMessage = () => {
		if (firstPin === '' || secondPin === '' || arePinEquals()) {
			return ''
		}
		return t(RegisterI18nKeys.ERROR)
	}

	return (
		<RegisterStyles.MainContainer>
			<StatusBar backgroundColor='transparent' style='auto' />
			<RegisterStyles.RecoveryButton
				onPress={() => navigation.navigate('Login')}
			>
				{t(RegisterI18nKeys.LOGIN)}
			</RegisterStyles.RecoveryButton>
			<RegisterStyles.Title>{t(RegisterI18nKeys.TITLE)}</RegisterStyles.Title>
			<RegisterStyles.Subtitle>{t(RegisterI18nKeys.SUBTITLE)}</RegisterStyles.Subtitle>

			<RegisterStyles.InputsContainer>
				<RegisterStyles.InputStyled
					disabled={isLoading}
					testID={testID.FIRST_PIN_INPUT}
					value={firstPin}
					keyboardType='decimal-pad'
					secureTextEntry
					onChangeText={(value: string) => {
						setFirstPin(value.replace(/[^0-9]/g, ''))
					}}
					placeholder={t(RegisterI18nKeys.PIN) || ''}
					returnKeyType={'next'}
				/>
				<RegisterStyles.InputStyled
					disabled={isLoading}
					testID={testID.SECOND_PIN_INPUT}
					value={secondPin}
					keyboardType='decimal-pad'
					secureTextEntry
					onChangeText={(value: string) => {
						setSecondPin(value.replace(/[^0-9]/g, ''))
					}}
					placeholder={t(RegisterI18nKeys.SECOND_PIN) || ''}
					returnKeyType={'done'}
				/>
				{renderErrorMessage() != '' && (
					<RegisterStyles.ErrorText testID={testID.REGISTER_ERROR_MESSAGE}>
						{renderErrorMessage()}
					</RegisterStyles.ErrorText>
				)}
				<RegisterStyles.CheckBoxContainer>
					<RegisterStyles.TermsCheckBox
						testID={testID.TERMS_CHECKBOX}
						checked={isTermsCheckboxChecked}
						onPress={() => {
							setIsTermsCheckboxChecked(!isTermsCheckboxChecked)
							!isTermsCheckboxChecked &&
								showModal?.({
									Component: TermsModal,
									modalContainerStyle: { paddingBottom: 40 },
								})
						}}

					/>
					<RegisterStyles.TermsText onPress={() => {
						showModal?.({
							Component: TermsModal,
							modalContainerStyle: { paddingBottom: 40 },
						})
						setIsTermsCheckboxChecked(true)
					}
					}>
						{t(RegisterI18nKeys.CONDITIONS)}
					</RegisterStyles.TermsText>
				</RegisterStyles.CheckBoxContainer>
			</RegisterStyles.InputsContainer>


			<RegisterStyles.ContainerBottom>
				{isLoading ? <RegisterStyles.SplashActivityIndicator color={getThemeColor(ColorKeys.text)} size={35} /> : null}
				<RegisterStyles.Button
					testID={testID.REGISTER_BUTTON}
					disabled={isPinFilled || !isTermsCheckboxChecked || isLoading}
					onPress={() => {
						setIsLoading(true)
					}}
				>
					{isLoading ? t('LOADING') : t('BUTTON')}
				</RegisterStyles.Button>
			</RegisterStyles.ContainerBottom>
		</RegisterStyles.MainContainer>
	)
}

export default Register
