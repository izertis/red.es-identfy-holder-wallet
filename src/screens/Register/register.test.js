import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import "@testing-library/jest-native/extend-expect";
import App from ".";
import { testID } from "./constants/testID";
import i18next from "../../../i18n.config";
import { SCREEN } from "../../constants/screens";
import es from "./i18n/es";

// describe block that groups together related tests for the <App /> component
describe("<Register />", () => {

  const mockShowModal = jest.fn();
  const ScreenName = SCREEN.Register;

  // Adds a resource bundle for the Spanish language to i18next with the translations for the Register screen
  i18next.addResourceBundle("es", ScreenName, es);

  // A mock navigation object used to simulate navigation actions
  const navigation = { navigate: jest.fn() };

  // Test to check if the component renders correctly
  it("renders correctly", () => {
    const { toJSON } = render(
      <App navigation={navigation} translation={i18next} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  // Test to check if the registration button is disabled when the pins are not equal or the terms checkbox is not checked
  it("disables the button if the pins are not equal or the terms checkbox is not checked", () => {
    const { getByTestId } = render(
      <App navigation={navigation} translation={i18next} />
    );
    const button = getByTestId(testID.REGISTER_BUTTON);
    const firstPinInput = getByTestId(testID.FIRST_PIN_INPUT);
    const secondPinInput = getByTestId(testID.SECOND_PIN_INPUT);
    const checkbox = getByTestId(testID.TERMS_CHECKBOX);

    // Simulates user input by changing the text of the pin inputs and clicking the terms checkbox
    fireEvent.changeText(firstPinInput, "1234");
    fireEvent.changeText(secondPinInput, "5678");
    fireEvent.press(checkbox);

    expect(button).toBeDisabled();

    // Simulates user input by changing the text of the second pin input to be equal to the first pin input
    fireEvent.changeText(secondPinInput, "1234");

    // Checks if the registration button is enabled when the pins are equal and the checkbox is checked
    expect(button).toBeEnabled();

    fireEvent.press(checkbox);
    expect(button).toBeDisabled();
  });

  // Test to check if an error message is displayed when the pins are not equal
  it("renders an error message if the pins are not equal", () => {
    const { getByTestId, queryAllByTestId } = render(
      <App navigation={navigation} translation={i18next} />
    );
    const firstPinInput = getByTestId(testID.FIRST_PIN_INPUT);
    const secondPinInput = getByTestId(testID.SECOND_PIN_INPUT);

    // Simulates user input by changing the text of the pin inputs to be different
    fireEvent.changeText(firstPinInput, "1234");
    fireEvent.changeText(secondPinInput, "5678");

    // Checks if an error message is displayed
    let messageErrorText = queryAllByTestId(testID.REGISTER_ERROR_MESSAGE);
    expect(messageErrorText).toHaveLength(1);

    // Set equal pins and ensure the error message is not displayed
    fireEvent.changeText(secondPinInput, "1234");
    messageErrorText = queryAllByTestId(testID.REGISTER_ERROR_MESSAGE);
    expect(messageErrorText).toHaveLength(0);
  });
});
