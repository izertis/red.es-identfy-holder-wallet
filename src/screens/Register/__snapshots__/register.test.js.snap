// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "height": "100%",
        "marginLeft": 25,
        "marginRight": 25,
        "marginTop": 15,
        "paddingBottom": 35,
        "paddingTop": 19,
      },
    ]
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "backgroundColor": "transparent",
        "borderRadius": 20,
        "marginBottom": 19,
        "marginLeft": "-2%",
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
    testID="button-container-outer-layer"
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "transparent",
          "borderColor": "transparent",
          "borderRadius": 20,
          "borderStyle": "solid",
          "borderWidth": 0,
          "flex": undefined,
          "minWidth": 64,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="button-container"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 20,
            },
          ]
        }
        testID="button"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "width": 120,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "marginLeft": 12,
                  "marginRight": -4,
                },
                Object {
                  "marginLeft": 16,
                  "marginRight": -16,
                },
                Object {
                  "marginLeft": 12,
                  "marginRight": -8,
                },
              ]
            }
            testID="button-icon-container"
          >
            <Text />
          </View>
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "textAlign": "left",
                },
                Object {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  Array [
                    Object {
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    false,
                    Object {
                      "marginHorizontal": 16,
                    },
                    undefined,
                    false,
                    Object {
                      "color": "rgba(103, 80, 164, 1)",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    Object {
                      "color": "#eb5b5b",
                      "fontFamily": "Muli-600",
                    },
                  ],
                ],
              ]
            }
            testID="button-text"
          >
            LOGIN
          </Text>
        </View>
      </View>
    </View>
  </View>
  <Text
    bold={true}
    className="sc-bcXHqe sc-hLBbgP sc-iBYQkv kFXtiY RwpaV"
    fontFamily="Montserrat"
    style={
      Array [
        Object {
          "textAlign": "left",
        },
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    TITLE
  </Text>
  <Text
    className="sc-bcXHqe sc-dkrFOg sc-ftTHYK laZgtI bNRHGr bUmJIx"
    style={
      Array [
        Object {
          "textAlign": "left",
        },
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    SUBTITLE
  </Text>
  <RCTScrollView
    style={
      Array [
        Object {},
      ]
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
            },
            undefined,
          ]
        }
      >
        <View
          style={Object {}}
        >
          <View
            pointerEvents="none"
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                },
                Object {
                  "backgroundColor": "rgba(255, 251, 254, 1)",
                  "borderColor": "#363d58",
                  "borderRadius": 4,
                  "borderWidth": 1,
                },
                undefined,
              ]
            }
            testID="text-input-outline"
          />
          <View>
            <View
              style={
                Array [
                  Object {
                    "paddingBottom": 0,
                  },
                  Object {
                    "minHeight": 56,
                    "paddingTop": 8,
                  },
                ]
              }
            >
              <TextInput
                className="sc-gKPRtg sc-pyfCe kwDacX fTbXGf"
                containerStyle={
                  Object {
                    "paddingHorizontal": 0,
                  }
                }
                cursorColor="#363d58"
                editable={true}
                errorStyle={
                  Object {
                    "color": "#eb5b5b",
                  }
                }
                inputContainerStyle={
                  Object {
                    "marginTop": 10,
                  }
                }
                inputStyle={
                  Object {
                    "fontSize": 16,
                  }
                }
                keyboardType="decimal-pad"
                maxFontSizeMultiplier={1.5}
                multiline={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="PIN"
                placeholderTextColor="#aaa"
                returnKeyType="next"
                secureTextEntry={true}
                selectionColor="#eb5b5b"
                style={
                  Array [
                    Object {
                      "margin": 0,
                      "zIndex": 1,
                    },
                    Object {
                      "height": 48,
                    },
                    Object {
                      "paddingBottom": 0,
                      "paddingTop": 0,
                    },
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "letterSpacing": 0.15,
                      "lineHeight": undefined,
                      "paddingHorizontal": 16,
                      "textAlign": "left",
                      "textAlignVertical": "center",
                    },
                    false,
                    Object {
                      "marginRight": 56,
                      "paddingRight": 0,
                    },
                    Object {
                      "fontFamily": "Muli",
                      "lineHeight": 25,
                    },
                  ]
                }
                testID="first-pin-input"
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "height": 24,
                    "justifyContent": "center",
                    "position": "absolute",
                    "width": 24,
                  },
                  Object {
                    "right": 16,
                    "top": 19,
                  },
                ]
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 20,
                    "height": 40,
                    "margin": 0,
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                    "width": 40,
                  }
                }
                testID="right-icon-adornment-container-outer-layer"
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "borderColor": "rgba(121, 116, 126, 1)",
                      "borderRadius": 20,
                      "borderWidth": 0,
                      "elevation": 0,
                      "flex": 1,
                      "overflow": "hidden",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0,
                      "shadowRadius": 0,
                    }
                  }
                  testID="right-icon-adornment-container"
                >
                  <View
                    accessibilityComponentType="button"
                    accessibilityRole="button"
                    accessibilityState={
                      Object {
                        "disabled": false,
                      }
                    }
                    accessibilityTraits="button"
                    accessible={true}
                    centered={true}
                    collapsable={false}
                    focusable={true}
                    hitSlop={
                      Object {
                        "bottom": 6,
                        "left": 6,
                        "right": 6,
                        "top": 6,
                      }
                    }
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "overflow": "hidden",
                        },
                        Array [
                          Object {
                            "alignItems": "center",
                            "flexGrow": 1,
                            "justifyContent": "center",
                          },
                          Object {
                            "borderRadius": 20,
                          },
                        ],
                      ]
                    }
                    testID="right-icon-adornment"
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
            },
            undefined,
          ]
        }
      >
        <View
          style={Object {}}
        >
          <View
            pointerEvents="none"
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                },
                Object {
                  "backgroundColor": "rgba(255, 251, 254, 1)",
                  "borderColor": "#363d58",
                  "borderRadius": 4,
                  "borderWidth": 1,
                },
                undefined,
              ]
            }
            testID="text-input-outline"
          />
          <View>
            <View
              style={
                Array [
                  Object {
                    "paddingBottom": 0,
                  },
                  Object {
                    "minHeight": 56,
                    "paddingTop": 8,
                  },
                ]
              }
            >
              <TextInput
                className="sc-gKPRtg sc-pyfCe kwDacX fTbXGf"
                containerStyle={
                  Object {
                    "paddingHorizontal": 0,
                  }
                }
                cursorColor="#363d58"
                editable={true}
                errorStyle={
                  Object {
                    "color": "#eb5b5b",
                  }
                }
                inputContainerStyle={
                  Object {
                    "marginTop": 10,
                  }
                }
                inputStyle={
                  Object {
                    "fontSize": 16,
                  }
                }
                keyboardType="decimal-pad"
                maxFontSizeMultiplier={1.5}
                multiline={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="SECOND_PIN"
                placeholderTextColor="#aaa"
                returnKeyType="done"
                secureTextEntry={true}
                selectionColor="#eb5b5b"
                style={
                  Array [
                    Object {
                      "margin": 0,
                      "zIndex": 1,
                    },
                    Object {
                      "height": 48,
                    },
                    Object {
                      "paddingBottom": 0,
                      "paddingTop": 0,
                    },
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "letterSpacing": 0.15,
                      "lineHeight": undefined,
                      "paddingHorizontal": 16,
                      "textAlign": "left",
                      "textAlignVertical": "center",
                    },
                    false,
                    Object {
                      "marginRight": 56,
                      "paddingRight": 0,
                    },
                    Object {
                      "fontFamily": "Muli",
                      "lineHeight": 25,
                    },
                  ]
                }
                testID="second-pin-input"
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "height": 24,
                    "justifyContent": "center",
                    "position": "absolute",
                    "width": 24,
                  },
                  Object {
                    "right": 16,
                    "top": 19,
                  },
                ]
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 20,
                    "height": 40,
                    "margin": 0,
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                    "width": 40,
                  }
                }
                testID="right-icon-adornment-container-outer-layer"
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "borderColor": "rgba(121, 116, 126, 1)",
                      "borderRadius": 20,
                      "borderWidth": 0,
                      "elevation": 0,
                      "flex": 1,
                      "overflow": "hidden",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0,
                      "shadowRadius": 0,
                    }
                  }
                  testID="right-icon-adornment-container"
                >
                  <View
                    accessibilityComponentType="button"
                    accessibilityRole="button"
                    accessibilityState={
                      Object {
                        "disabled": false,
                      }
                    }
                    accessibilityTraits="button"
                    accessible={true}
                    centered={true}
                    collapsable={false}
                    focusable={true}
                    hitSlop={
                      Object {
                        "bottom": 6,
                        "left": 6,
                        "right": 6,
                        "top": 6,
                      }
                    }
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "overflow": "hidden",
                        },
                        Array [
                          Object {
                            "alignItems": "center",
                            "flexGrow": 1,
                            "justifyContent": "center",
                          },
                          Object {
                            "borderRadius": 20,
                          },
                        ],
                      ]
                    }
                    testID="right-icon-adornment"
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "paddingTop": 24,
            },
          ]
        }
      >
        <View
          accessibilityLiveRegion="polite"
          accessibilityRole="checkbox"
          accessibilityState={
            Object {
              "checked": false,
              "disabled": false,
            }
          }
          accessible={true}
          checked={false}
          collapsable={false}
          color="#363d58"
          containerStyle={
            Object {
              "marginHorizontal": 5,
              "padding": 4,
            }
          }
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 18,
                "height": 36,
                "padding": 6,
                "width": 36,
              },
            ]
          }
          testID="terms-checkbox"
        >
          <View
            collapsable={false}
            style={
              Object {
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <Text />
            <View
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "borderColor": "rgba(73, 69, 79, 1)",
                    "borderWidth": 0,
                    "height": 14,
                    "width": 14,
                  }
                }
              />
            </View>
          </View>
        </View>
        <Text
          className="sc-bcXHqe sc-dkrFOg sc-jrcTuL laZgtI bNRHGr cShEYu"
          onPress={[Function]}
          style={
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
                "letterSpacing": 0,
              },
              Object {
                "writingDirection": "ltr",
              },
              undefined,
            ]
          }
        >
          CONDITIONS
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "flex-end",
          "marginBottom": 40,
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": 20,
          "width": "100%",
        },
      ]
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 20,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="register-button-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 20,
            "borderStyle": "solid",
            "borderWidth": 0,
            "flex": undefined,
            "minWidth": 64,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="register-button-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 20,
              },
            ]
          }
          testID="register-button"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                Object {
                  "backgroundColor": "#ccc",
                  "borderRadius": 15,
                  "fontFamily": "Muli",
                  "height": 39,
                  "width": 224,
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    Array [
                      Object {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      Object {
                        "marginHorizontal": 12,
                      },
                      undefined,
                      false,
                      Object {
                        "color": "rgba(28, 27, 31, 0.38)",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      Object {
                        "color": "#fff",
                        "fontFamily": "Muli",
                        "letterSpacing": 0.57,
                      },
                    ],
                  ],
                ]
              }
              testID="register-button-text"
            >
              BUTTON
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
